docs/ideas/yaml/jmespath.toml

```diff
@@ -0,0 +1,89 @@
+# The values below were taken from .github/workflows/python.yaml in this repo
+
+# 1. JMESPath as part of the section name, after the file name.
+# Everything after the file name is considered a JMESPath https://jmespath.org/
+# Format: ["path/to/file.ext".jmes.path.expression]
+#
+# 2. "jobs.build.strategy.matrix" should have "os" and "python-version"
+# 3. Both are lists, and they have to be exactly as described here.
+[".github/workflows/python.yaml".jobs.build.strategy.matrix]
```

Comment:
Maybe also allow something like:
```toml
[".github/workflows/python.yaml"]
"jobs.build.strategy.matrix.os" = "foo"
"jobs.build.steps" = ["bar"]
"jobs.build.steps.regex" = "baz \d+"
"jobs.build.steps.contains" = "baz"
```

The last three would not appear at the same time obviously..

Reply:
That reads quite nicely but how do you know the keys are JMESPath in that case?

Reply:
No clue :-) I wrote that from the perspective of a naive user. Not sure if it is implementable

Reply:
Yes, I used the wrong point-of-view.
I second your suggestion :)